src/main/scala/wdl4s/wom/graph/ScatterNode.scala

```diff
@@ -0,0 +1,58 @@
+package wdl4s.wom.graph
+
+import wdl4s.wom.graph.GraphNodePort.{ConnectedInputPort, InputPort, OutputPort, ScatterGathererPort}
+import wdl4s.wom.graph.ScatterNode.ScatterVariableMapping
+import wdl4s.wdl.types.WdlArrayType
+
+final case class ScatterNode private(graph: Graph,
+                                     scatterVariableMapping: ScatterVariableMapping,
+                                     otherInputPorts: Set[InputPort],
+                                     outputMapping: Set[ScatterGathererPort]) extends GraphNode {
+
+  def innerGraphInputs: Set[GraphInputNode] = graph.nodes.collect { case gin: GraphInputNode => gin }
+
+  // NB if you find yourself calling .filter on this set of inputPorts, you probably just wanted to access either
+  // the scatterVariableMapping or otherInputPorts fields directly.
```

Comment:
Imagine that the `scatter` block were a self-contained unit and you had to write an inputs JSON for it... It's **that** set minus the scatter variable.